### PR TITLE
Add a new TiDB Cloud Integration

### PR DESCRIPTION
**Pull request by SabaPing**
### What does this PR do?

Add a new integration for [TiDB Cloud](https://tidbcloud.com/).

### Motivation

Previously, I submitted the [TiDB Integration](https://docs.datadoghq.com/integrations/tidb/) for our on-premises users. Some of our TiDB Cloud users want the similar functionality. 

So there is this integration.

> This integration is a tile integration only.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [ ] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### CODE REVIEW

**Comment by SabaPing** [comment]
Since there are more work to do, I turn this PR to `draft` state for now.

**Comment by urseberry** [comment]
> Since there are more work to do, I turn this PR to `draft` state for now.

Thank you! Please let us know when it is ready for review.

**Comment by SabaPing** [comment]
I've done development from TiDBCloud side. This PR is ready for review now.
Thanks a lot!

File: tidb_cloud/metadata.csv [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+tidb_cloud.db_query_duration_seconds.count,count,,query,,Total number of handled queries in server,1,tidb_cloud,
+tidb_cloud.db_query_duration_seconds.sum,count,,second,,Sum of handled query duration in server,1,tidb_cloud,
+tidb_cloud.storage_used_bytes,gauge,,byte,,Disk usage bytes of TiKV/TiFlash instances,1,tidb_cloud,
+tidb_cloud.storage_capacity_bytes,gauge,,byte,,Disk capacity bytes of TiKV/TiFlash instances,1,tidb_cloud,
+tidb_cloud.storage_tikv_io_bytes_total,count,,byte,,IO read/write bytes of TiKV instances,1,tidb_cloud,
+tidb_cloud.process_cpu_seconds_total,count,,second,,Sum of CPU usage seconds of TiDB/TiKV/TiFlash instances,1,tidb_cloud,
+tidb_cloud.process_cpu_used_cores,gauge,,core,,CPU usage cores of TiDB/TiKV/TiFlash instances,1,tidb_cloud,
+tidb_cloud.node_cpu_capacity_cores,gauge,,core,,CPU limit cores of TiDB/TiKV/TiFlash instances,1,tidb_cloud,
+tidb_cloud.process_memory_used_bytes,gauge,,byte,,Resident memory bytes of TiDB/TiKV/TiFlash instances,1,tidb_cloud,
+tidb_cloud.node_memory_capacity_bytes,gauge,,byte,,Memory limit bytes of TiDB/TiKV/TiFlash instances,1,tidb_cloud,
```
Comment (coignetp) [inline tidb_cloud/metadata.csv:14]:
```suggestion
tidb_cloud.db_queries_total,count,,query,,Total number of statements executed,0,tidb_cloud,
tidb_cloud.db_failed_queries_total,count,,query,,Total number of execution errors,0,tidb_cloud,
tidb_cloud.db_connections,gauge,,connection,,Current number of connections in TiDB server,0,tidb_cloud,
tidb_cloud.db_query_duration_seconds.count,count,,query,,Total number of handled queries in server,0,tidb_cloud,
tidb_cloud.db_query_duration_seconds.sum,count,,second,,Sum of handled query duration in server,0,tidb_cloud,
tidb_cloud.storage_used_bytes,gauge,,byte,,Disk usage bytes of TiKV/TiFlash instances,0,tidb_cloud,
tidb_cloud.storage_capacity_bytes,gauge,,byte,,Disk capacity bytes of TiKV/TiFlash instances,0,tidb_cloud,
tidb_cloud.storage_tikv_io_bytes_total,count,,byte,,IO read/write bytes of TiKV instances,0,tidb_cloud,
tidb_cloud.process_cpu_seconds_total,count,,second,,Sum of CPU usage seconds of TiDB/TiKV/TiFlash instances,0,tidb_cloud,
tidb_cloud.process_cpu_used_cores,gauge,,core,,CPU usage cores of TiDB/TiKV/TiFlash instances,0,tidb_cloud,
tidb_cloud.node_cpu_capacity_cores,gauge,,core,,CPU limit cores of TiDB/TiKV/TiFlash instances,0,tidb_cloud,
tidb_cloud.process_memory_used_bytes,gauge,,byte,,Resident memory bytes of TiDB/TiKV/TiFlash instances,0,tidb_cloud,
tidb_cloud.node_memory_capacity_bytes,gauge,,byte,,Memory limit bytes of TiDB/TiKV/TiFlash instances,0,tidb_cloud,
```

**Review by coignetp — CHANGES_REQUESTED** [review]
Thanks! Looks good from agent-integrations side, just a tiny suggestion.
Pinging @DataDog/documentation for a review

File: tidb_cloud/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# TiDB Cloud Integration
+
+## Overview
+
+[TiDB Cloud][1] makes deploying, managing and maintaining your TiDB clusters even simpler with a fully managed cloud instance that you control through an intuitive console. 
+
+TiDB Cloud allows developers and DBAs with little or no training to handle once-complex tasks such as infrastructure management and cluster deployment with ease, to focus on your applications, not the complexities of your database.
+
+The Datadog integration is an observability feature of TiDB Cloud that exports metrics of TiDB Cloud clusters to your Datadog platform.
```
Comment (urseberry) [inline tidb_cloud/README.md:9]:
```suggestion
[TiDB Cloud][1] is a fully managed cloud service of TiDB, an open-source database.

Use the TiDB Cloud Datadog integration to export metrics from TiDB Cloud clusters to Datadog.
```

File: tidb_cloud/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+> **Note:**
+>
+> - For TiDB clusters on premises, please refer to [TiDB Integration][4].
+
+## Setup
+
+To set up the TiDB Cloud Datadog integration for your cluster, provide a Datadog API key and region to TiDB Cloud.
+
+Refer to the [TiDB Cloud Preferences Page][2] to configure the Datadog integration for your TiDB Cloud project.
```
Comment (urseberry) [inline tidb_cloud/README.md:19]:
```suggestion
See [TiDB Cloud Preferences][2] to configure the Datadog integration for your TiDB Cloud project.
```

**Review by urseberry — COMMENTED** [review]
Hi, thanks for writing this up and submitting it!

- I believe you can remove the empty `images/.gitkeep` file
- There is a note in the main TiDB integrations Overview section saying that the Datadog integration for TiDB Cloud is unavailable. I assume that line is now out of date. Will you submit a PR to update it?

File: tidb_cloud/README.md (hunk 2 above)
Comment (urseberry) [reply tidb_cloud/README.md:9]: Sorry for removing your writing, but we don't allow marking-type prose in the integrations docs.

**Comment by SabaPing** [comment]
I have added a reference from TiDB Integration here.
https://github.com/DataDog/integrations-extras/pull/1094/files#diff-e1bf31e4f606c5e3e812ffff04524dcaeb8e108a08bc810fc3920d572920332eR14

**Review by coignetp — APPROVED** [review]
Looks good! Needs a final stamp from documentation before merging

**Review by urseberry — APPROVED** [review]

**Pull request merged**